owncloudComLibrary/src/main/java/com/owncloud/android/lib/common/utils/Log_OC.java

```diff
@@ -1,212 +1,52 @@
-            Log.e("OC_Log", "Stopping logging failed: ", e);
-        }
+        Timber.w(message);
     }
 
-    /**
-     * Delete history logging
-     */
-    public static void deleteHistoryLogging() {
-        File folderLogs = new File(mFolder + File.separator);
-        if (folderLogs.isDirectory()) {
-            String[] myFiles = folderLogs.list();
-            for (String fileName : myFiles) {
-                File fileInFolder = new File(folderLogs, fileName);
-                Log_OC.d("delete file", fileInFolder.getAbsoluteFile() + " " + fileInFolder.delete());
-            }
-        }
+    public static void startLogging(String storagePath) {
+        LoggingHelper.INSTANCE.startLogging(
+                new File(storagePath+ File.separator + mOwncloudDataFolderLog), mOwncloudDataFolderLog);
     }
 
-    /**
-     * Append the info of the device
-     */
-    private static void appendPhoneInfo() {
-        appendLog("Model : " + android.os.Build.MODEL);
-        appendLog("Brand : " + android.os.Build.BRAND);
-        appendLog("Product : " + android.os.Build.PRODUCT);
-        appendLog("Device : " + android.os.Build.DEVICE);
-        appendLog("Version-Codename : " + android.os.Build.VERSION.CODENAME);
-        appendLog("Version-Release : " + android.os.Build.VERSION.RELEASE);
+    public static void stopLogging() {
+        LoggingHelper.INSTANCE.stopLogging();
     }
 
-    /**
-     * Append to the log file the info passed
-     *
-     * @param text : text for adding to the log file
-     */
-    synchronized private static void appendLog(String text) {
-
-        if (isEnabled) {
-
-            if (isMaxFileSizeReached) {
-
-                // Move current log file info to another file (old logs)
-                File olderFile = new File(mFolder + File.separator + mLogFileNames[1]);
-                if (mLogFile.exists()) {
-                    mLogFile.renameTo(olderFile);
-                }
-
-                // Construct a new file for current log info
-                mLogFile = new File(mFolder + File.separator + mLogFileNames[0]);
-                isMaxFileSizeReached = false;
-            }
-
-            String timeStamp = new SimpleDateFormat(SIMPLE_DATE_FORMAT, Locale.ENGLISH).format(Calendar.getInstance().getTime());
-
-            try {
-                mBuf = new BufferedWriter(new FileWriter(mLogFile, true));
-                mBuf.newLine();
-                mBuf.write(timeStamp + " " + text);
-            } catch (IOException e) {
-                e.printStackTrace();
-            } finally {
-                try {
-                    mBuf.close();
-                } catch (IOException e) {
-                    e.printStackTrace();
-                }
-            }
-
-            // Check if current log file size is bigger than the max file size defined
-            if (mLogFile.length() > MAX_FILE_SIZE) {
-                isMaxFileSizeReached = true;
-            }
```

Comment:
What happens if we lose this check? Could logs be huge or how is logs size controlled now?

Reply:
### Before
There were two files. Old and new. The new overwrites the old. So it means overall log were split into two similar size files and you see always the current. 

### New
EveryTime you run into Application create a new file with current date is created/reused. The old keeps there till you manually remove it. 
Btw, the file-logging is in my point of view pointless, you see current logcat (in `release` !) anyways. It was a workaround to see what happens in `release` anyhow

Reply:
Ok, thanks for this explanation! I think this is something to check with @davigonz and @jesmrec

Reply:
I will check in QA stage. Just one question: does the rotation happen every time you open the app? does this not lead to log file flooding?

Reply:
No, only if it's really opened. Only a click on the icon does not start app, when it's already open in background.
But when you remove app from task list, it will create a file, but maximum only one per day